server/config.go

```diff
@@ -70,21 +83,53 @@ func (s *Service) ConfigSection(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	section := httprouter.GetParamFromContext(ctx, "section")
 	var res interface{}
-	switch section {
-	case "auth":
-		res = newAuthConfigResponse(*config)
-	default:
-		Error(w, http.StatusBadRequest, fmt.Sprintf("received unknown section %q", section), s.Logger)
+	res = newLogViewerUIConfigResponse(*config)
+
+	encodeJSON(w, http.StatusOK, res, s.Logger)
+}
+
+// ReplaceLogViewerUIConfig replaces the log viewer UI section of the global application configuration
+func (s *Service) ReplaceLogViewerUIConfig(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	config, err := s.Store.Config(ctx).Get(ctx)
+	if err != nil {
+		Error(w, http.StatusBadRequest, err.Error(), s.Logger)
+		return
+	}
+
+	if config == nil {
+		Error(w, http.StatusBadRequest, "Configuration object was nil", s.Logger)
+		return
+	}
+
+	var res interface{}
+
+	var logViewerUIConfig chronograf.LogViewerUIConfig
+	if err := json.NewDecoder(r.Body).Decode(&logViewerUIConfig); err != nil {
```

Comment:
Why do we retrieve the config before decoding the request body?

Reply:
do you think we should decode and validate before accessing the database?

Reply:
yup

Reply:
:thu